### PR TITLE
feat!: Remove deprecated ‘in list’ variant of Link

### PR DESCRIPTION
**Pull request by VincentSmedinga**
We deprecated this value for the `variant` prop of List in yesterday’s release. This PR removes it for the next release.

It also updates all our examples – something we should actually have done with the deprecation. This includes making the links in the Footer small, and the primary ones in the Mega Menu large. Sorry for not separating commits.

### CODE REVIEW

File: packages/react/src/Link/Link.tsx
Comment (dlnr) [inline packages/react/src/Link/Link.tsx:1]: You've opted to use jsdocs everywhere. Should we add them if we touch a file without jsdocs?

**Review by dlnr — APPROVED** [review]

File: packages/react/src/Link/Link.tsx
Comment (VincentSmedinga) [reply packages/react/src/Link/Link.tsx:1]: Yes, but let’s first discuss how we want to write this aspect of documentation.

File: storybook/storybook-react/src/MegaMenu/MegaMenu.stories.tsx [hunk 1]
```diff
@@ -156,147 +76,79 @@ export const MultipleCategories: Story = {
...
-                </Link>
-              </UnorderedList.Item>
-              <UnorderedList.Item>
-                <Link variant="inList" href="#">
-                  Subsidies
-                </Link>
-              </UnorderedList.Item>
-            </UnorderedList>
+            <LinkList>
+              <LinkList.Link href="#" size="large">
```
Comment (alimpens) [inline storybook/storybook-react/src/MegaMenu/MegaMenu.stories.tsx:83]: Out of scope for this PR, but I missed it in the LinkList PR: do we really want a `large` link variant? It looks a little off imo

**Review by alimpens — APPROVED** [review]

File: storybook/storybook-react/src/MegaMenu/MegaMenu.stories.tsx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/storybook-react/src/MegaMenu/MegaMenu.stories.tsx:83]: Noted for discussion.

**Pull request merged**